File: examples/js/langchain-chat.mjs
// LangChain.js against FreeTheAi.
//
// Usage:
//   npm install langchain @langchain/openai @langchain/core
//   export FREETHEAI_API_KEY=fta_...
//   node js/langchain-chat.mjs

import { ChatOpenAI } from "@langchain/openai";
import { HumanMessage, SystemMessage } from "@langchain/core/messages";

const apiKey = process.env.FREETHEAI_API_KEY;
if (!apiKey) {
    throw new Error("Set FREETHEAI_API_KEY before running this example.");
}

const llm = new ChatOpenAI({
    model: "bbg/zai-org/GLM-5.1",
    apiKey,
    configuration: {
        baseURL: "https://api.freetheai.xyz/v1",
    },
    maxTokens: 128,
});

const response = await llm.invoke([
    new SystemMessage("You are concise."),
    new HumanMessage("Reply with exactly: OK"),
]);

console.log(response.content);

File: examples/python/langchain_chat.py
#!/usr/bin/env python3
"""LangChain against FreeTheAi.

Uses the standard `ChatOpenAI` integration with a custom `base_url`.

Usage:
    pip install langchain langchain-openai
    export FREETHEAI_API_KEY=fta_...
    python python/langchain_chat.py
"""
from __future__ import annotations

import os

from langchain_core.messages import HumanMessage, SystemMessage
from langchain_openai import ChatOpenAI


def main() -> None:
    """
    var: result
    type: None
    desc: Run a single LangChain prompt against FreeTheAi.
    """
    key = os.environ.get("FREETHEAI_API_KEY")
    if not key:
        raise RuntimeError("Set FREETHEAI_API_KEY before running this example.")

    llm = ChatOpenAI(
        model="bbg/zai-org/GLM-5.1",
        api_key=key,
        base_url="https://api.freetheai.xyz/v1",
        max_tokens=128,
    )
    response = llm.invoke([
        SystemMessage(content="You are concise."),
        HumanMessage(content="Reply with exactly: OK"),
    ])
    print(response.content)


if __name__ == "__main__":
    main()
